src/backend.c

```diff
@@ -128,11 +128,16 @@ lookup_backend(const struct Backend_head *head, const char *name, size_t name_le
 
     STAILQ_FOREACH(iter, head, entries) {
         assert(iter->pattern_re != NULL);
-        if (pcre_exec(iter->pattern_re, NULL,
-                    name, name_len, 0, 0, NULL, 0) >= 0)
+        md = pcre2_match_data_create_from_pattern(iter->pattern_re,NULL);
```

Comment:
When there are multiple entries in the list this will leak memory.

Reply:
I think this is now resolved by moving the call to init_backend and the free to remove_backend.